---
int32 error
int32 system_status